### PR TITLE
SQS-379 | unit tests for processOrderBookActiveOrders

### PR DESCRIPTION
**Pull request by deividaspetraitis**
Implements tests for processOrderBookActiveOrders orderbook usecase method. As a side effect introduces some helper methods and refactors some test cases.

### CODE REVIEW

File: orderbook/usecase/orderbook_usecase_test.go [hunk 1]
```diff
@@ -401,42 +457,144 @@ func (s *OrderbookUsecaseTestSuite) TestGetActiveOrders() {
 	}
 }
 
-func (s *OrderbookUsecaseTestSuite) TestCreateFormattedLimitOrder() {
-	getTickByIDFunc := func(effectiveTotalAmountSwapped string, unrealizedCancels int64, direction string) func(poolID uint64, tickID int64) (orderbookdomain.OrderbookTick, bool) {
-		return func(poolID uint64, tickID int64) (orderbookdomain.OrderbookTick, bool) {
-			tickValues := orderbookdomain.TickValues{
-				EffectiveTotalAmountSwapped: effectiveTotalAmountSwapped,
+func (s *OrderbookUsecaseTestSuite) TestProcessOrderBookActiveOrders() {
+	withGetMetadataByChainDenom := func(tokensusecase *mocks.TokensUsecaseMock, errIfNotDenom string) *mocks.TokensUsecaseMock {
```
Comment (p0mvn) [inline orderbook/usecase/orderbook_usecase_test.go]: Some comments here would help to grok what's going on 🙏

File: orderbook/usecase/orderbook_usecase_test.go [hunk 2]
```diff
@@ -35,6 +35,101 @@ func TestOrderbookUsecaseTestSuite(t *testing.T) {
...
 
+// Creates a new osmomath.Dec from a string
+// Panics if the string is invalid
+func (s *OrderbookUsecaseTestSuite) newDecFromStr(str string) osmomath.Dec {
+	s.T().Helper()
+	dec, err := osmomath.NewDecFromStr(str)
+	s.Require().NoError(err)
+	return dec
+}
+
```
Comment (p0mvn) [inline orderbook/usecase/orderbook_usecase_test.go]: Consider directly using and removing the helper:
```go
osmomath.MustNewDecFromStr(...)
```

File: orderbook/usecase/orderbook_usecase_test.go [hunk 3]
```diff
@@ -35,6 +35,101 @@ func TestOrderbookUsecaseTestSuite(t *testing.T) {
...
+		OrderId:        1,
+		OrderDirection: "bid",
+		Owner:          "owner1",
+		Quantity:       "1000",
+		PlacedQuantity: "1500",
+		Etas:           "500",
+		ClaimBounty:    "10",
+		PlacedAt:       "1634764800000",
+	}
+}
```
Comment (p0mvn) [inline orderbook/usecase/orderbook_usecase_test.go]: Why do we need to define it as a method/helper?

Could we for simplicity define the variable at the top of the file instead?

i.e.
```go
var defaultOrder = orderbookdomain.Order{
  TickId:         1,
  OrderId:        1,
  OrderDirection: "bid",
  Owner:          "owner1",
  Quantity:       "1000",
  PlacedQuantity: "1500",
  Etas:           "500",
  ClaimBounty:    "10",
  PlacedAt:       "1634764800000",
}
```

File: orderbook/usecase/orderbook_usecase_test.go [hunk 4]
```diff
@@ -35,6 +35,101 @@ func TestOrderbookUsecaseTestSuite(t *testing.T) {
...
+
+func (s *OrderbookUsecaseTestSuite) newLimitOrder() orderbookdomain.LimitOrder {
```
Comment (p0mvn) [inline orderbook/usecase/orderbook_usecase_test.go]: Ditto about helper

File: orderbook/usecase/orderbook_usecase_test.go [hunk 5]
```diff
@@ -35,6 +35,101 @@ func TestOrderbookUsecaseTestSuite(t *testing.T) {
...
+		PercentFilled:    s.newDecFromStr("0.400000000000000000"),
+		OrderbookAddress: "someOrderbookAddress",
+		Status:           "partiallyFilled",
+		Output:           s.newDecFromStr("1499.998500001499998500"),
+		QuoteAsset:       orderbookdomain.Asset{Symbol: "ATOM", Decimals: 6},
+		BaseAsset:        orderbookdomain.Asset{Symbol: "OSMO", Decimals: 6},
+	}
+}
+
+func (s *OrderbookUsecaseTestSuite) newTick(effectiveTotalAmountSwapped string, unrealizedCancels int64, direction string) orderbookdomain.OrderbookTick {
```
Comment (p0mvn) [inline orderbook/usecase/orderbook_usecase_test.go]: This one makes sense to be a helper IMO - nice work!

However, could you please add some comments to make this more grokable?

File: orderbook/usecase/orderbook_usecase_test.go [hunk 6]
```diff
@@ -401,42 +457,144 @@ func (s *OrderbookUsecaseTestSuite) TestGetActiveOrders() {
...
+			} else {
+				s.Assert().NoError(err)
+				s.Assert().Equal(tc.expectedOrders, orders)
+				s.Assert().Equal(tc.expectedIsBestEffort, isBestEffort)
+			}
+		})
 	}
+}
 
+func (s *OrderbookUsecaseTestSuite) TestCreateFormattedLimitOrder() {
```
Comment (p0mvn) [inline orderbook/usecase/orderbook_usecase_test.go:612]: Nice clean up for this test.

File: orderbook/usecase/orderbook_usecase_test.go [hunk 7]
```diff
@@ -401,42 +457,144 @@ func (s *OrderbookUsecaseTestSuite) TestGetActiveOrders() {
...
+
+			order := s.newLimitOrder()
+
+			// Call the method under test
+			orders, isBestEffort, err := usecase.ProcessOrderBookActiveOrders(context.Background(), domain.CanonicalOrderBooksResult{
+				ContractAddress: order.OrderbookAddress,
+				PoolID:          1,
+				Quote:           order.QuoteAsset.Symbol,
+				Base:            order.BaseAsset.Symbol,
+			}, "ownerAddress")
```
Comment (p0mvn) [inline orderbook/usecase/orderbook_usecase_test.go]: Similar to comments in other PRs, I think we should have one test case where we vary the `ownderAddress`

File: orderbook/usecase/orderbook_usecase_test.go [hunk 8]
```diff
@@ -401,42 +457,144 @@ func (s *OrderbookUsecaseTestSuite) TestGetActiveOrders() {
...
+
+			// Assert the results
+			if tc.expectedError != nil {
+				s.Assert().Error(err)
+				if errors.Is(err, tc.expectedError) {
+					s.Assert().ErrorIs(err, tc.expectedError)
+				} else {
+					s.Assert().ErrorAs(err, tc.expectedError)
+				}
```
Comment (p0mvn) [inline orderbook/usecase/orderbook_usecase_test.go]: I've seen this logic in a previous PR review - I think it makes sense to move it into a helper defined on `routertesting` to reduce code duplication.

File: orderbook/usecase/orderbook_usecase_test.go [hunk 9]
```diff
@@ -401,42 +457,144 @@ func (s *OrderbookUsecaseTestSuite) TestGetActiveOrders() {
...
+			}
+
+			order := s.newLimitOrder()
+
+			// Call the method under test
+			orders, isBestEffort, err := usecase.ProcessOrderBookActiveOrders(context.Background(), domain.CanonicalOrderBooksResult{
+				ContractAddress: order.OrderbookAddress,
+				PoolID:          1,
+				Quote:           order.QuoteAsset.Symbol,
+				Base:            order.BaseAsset.Symbol,
```
Comment (p0mvn) [inline orderbook/usecase/orderbook_usecase_test.go]: Ditto for at least one case where test parameters are diverging from default to make sure that there is no logic, the success of which is coupled to a specific value of a parameter

File: orderbook/usecase/orderbook_usecase_test.go [hunk 10]
```diff
@@ -401,42 +457,144 @@ func (s *OrderbookUsecaseTestSuite) TestGetActiveOrders() {
...
+				client.GetActiveOrdersCb = s.getActiveOrdersFunc(orderbookdomain.Orders{s.newOrder()}, 1, nil)
+				withGetMetadataByChainDenom(tokensusecase, "")
+				orderbookrepository.GetTickByIDFunc = s.getTickByIDFunc(orderbookdomain.OrderbookTick{}, false)
+			},
+			expectedOrders:       []orderbookdomain.LimitOrder{},
+			expectedError:        nil,
+			expectedIsBestEffort: true,
+		},
+		{
+			name: "successful processing of active orders",
```
Comment (p0mvn) [inline orderbook/usecase/orderbook_usecase_test.go]: Would love to see 2 extra test cases:
- 2 successful orders processed
- 2 processed but one is best-effort

File: orderbook/usecase/orderbook_usecase_test.go (hunk 2 above)
Comment (deividaspetraitis) [reply orderbook/usecase/orderbook_usecase_test.go]: Updated accordingly :+1:

File: orderbook/usecase/orderbook_usecase_test.go (hunk 3 above)
Comment (deividaspetraitis) [reply orderbook/usecase/orderbook_usecase_test.go]: Just did that!

File: orderbook/usecase/orderbook_usecase_test.go (hunk 1 above)
Comment (deividaspetraitis) [reply orderbook/usecase/orderbook_usecase_test.go]: Added comments, please check [here](https://github.com/osmosis-labs/sqs/pull/499/files#diff-5fa55bd6410efa835405589edaffbb41ba95f810344564647d2d711e8bc15b67R191)! :pray:

File: orderbook/usecase/orderbook_usecase_test.go (hunk 5 above)
Comment (deividaspetraitis) [reply orderbook/usecase/orderbook_usecase_test.go]: Added some docs/comments :+1:

File: orderbook/usecase/orderbook_usecase_test.go (hunk 4 above)
Comment (deividaspetraitis) [reply orderbook/usecase/orderbook_usecase_test.go]: Moved to [OrderbookTestHelper](https://github.com/osmosis-labs/sqs/pull/499/files#diff-5fa55bd6410efa835405589edaffbb41ba95f810344564647d2d711e8bc15b67R123), please let me does it make sense.

File: orderbook/usecase/orderbook_usecase_test.go (hunk 9 above)
Comment (deividaspetraitis) [reply orderbook/usecase/orderbook_usecase_test.go]: I have [moved](https://github.com/osmosis-labs/sqs/pull/499/files#diff-5fa55bd6410efa835405589edaffbb41ba95f810344564647d2d711e8bc15b67R654) `order` as a test case parameter.  :+1:

File: orderbook/usecase/orderbook_usecase_test.go (hunk 10 above)
Comment (deividaspetraitis) [reply orderbook/usecase/orderbook_usecase_test.go]: For 2 orders created this [one](https://github.com/osmosis-labs/sqs/pull/499/files#diff-5fa55bd6410efa835405589edaffbb41ba95f810344564647d2d711e8bc15b67R749), and updated best [effort](https://github.com/osmosis-labs/sqs/pull/499/files#diff-5fa55bd6410efa835405589edaffbb41ba95f810344564647d2d711e8bc15b67R705).

**Review by p0mvn — APPROVED** [review]
Good work on test coverage and keeping the test readable!

**Pull request merged**